Add Job/Bridge Creation Page

NavyAdmiral opened the pull request:
[Addresses #157965400](https://www.pivotaltracker.com/story/show/157965400) && [#157965381](https://www.pivotaltracker.com/story/show/157965381)

For now the default user and pw are `chainlink: twochains`.
Creates new bridges/jobs successfully, but doesn't redirect anywhere yet. Should bridge creation always redirect to /bridges ? I think that would be an extra step that isn't required at all. Maybe render {Bridge X} created is just enough?
[There's probably bunch of stuff to fix and enhance, but here's all the new features I added:](https://i.imgur.com/TSVc4X7.gifv)
![gif](https://thumbs.gfycat.com/TinyCourteousHornedviper-size_restricted.gif)

@rupurt instead of using react-toastify for notifications we could use the ones from the faucet

NavyAdmiral commented:
@rupurt I added support for [#158042699](https://www.pivotaltracker.com/story/show/158042699) with the last commit. I didn't do a separate PR because that commit builds on this PR. Preview for #158042699:
![gif](https://thumbs.gfycat.com/BronzePresentKillifish-size_restricted.gif)

rupurt commented:
@NavyAdmiral Github has feature that lets you set the base branch which i think you should be able to use.

<img width="1054" alt="screen shot 2018-07-19 at 11 31 45 am" src="https://user-images.githubusercontent.com/680789/42962889-8df78d4e-8b47-11e8-95c1-1c78c840d0ba.png">

If that doesn't work, can you hold the changes for story back `#158042699` as it makes it more difficult to review the changes in the PR.

se3000 commented:
@NavyAdmiral at first glance this looks great, but completely untested. To ease long term maintenance, we try to get at least some test coverage around new features. Could you write a success and error test for each page? Thanks!

NavyAdmiral commented on gui/src/components/BridgeForm.js:
```diff
@@ -0,0 +1,107 @@
+      confirmations: confirmations || ''
+    }
+  },
+  validationSchema: Yup.object().shape({
```
Remove client validation and just render the error in a red banner after submission?

NavyAdmiral commented:
@rupurt I've addressed most of the review. I'll wait for your PR to add notifications (removed for now). 
I think it will be a better user experience if a notification banner shows up on successful creation with a <Link> to the created page. Multiple submissions will be prevented by a disabled button.

NavyAdmiral commented:
@rupurt Something strange is happening when this PR is rebased to master; react-static-build fails with this error:
`
  Error: Invariant Violation: Failed exporting HTML for URL create (undefined): Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to e  xport your component from the file it's defined in, or you might have mixed up default and named imports.
`
I don't understand why this is failing since this branch is building without errors. (even though it's 100 commits behind there are no conflicts/changes that would cause this error)

dimroc commented on gui/src/components/BridgeForm.js:
```diff
@@ -0,0 +1,89 @@
+  handleSubmit (values) {
+    const formattedValues = JSON.parse(JSON.stringify(values).replace('confirmations', 'defaultConfirmations'))
+    formattedValues.defaultConfirmations = parseInt(formattedValues.defaultConfirmations) || 0
+    postBridge(formattedValues)
```
You are calling `postBridge` directly on `api.js` rather than through redux in `action.js`. Any particular reason for breaking the paradigm set with the `fetch*` actions?

rupurt replied:
@NavyAdmiral agreed that we want to send these through redux actions. There's some upcoming work on a global loading indicator that will leverage it.

rupurt commented:
@NavyAdmiral PR with flash banner is up #467

NavyAdmiral commented:
This PR has is somewhat bloated, so closing in the favor of a cleaner one.
